ycmd/completers/general/filename_completer.py

```diff
@@ -133,19 +138,39 @@ def GetPathsIncludeCase( self, path_dir, include_current_file_dir, filepath,
     return sorted( set( paths ) )
 
 
-def _GetPathsStandardCase( path_dir, use_working_dir, filepath ):
-  if not use_working_dir and not path_dir.startswith( '/' ):
-    path_dir = os.path.join( os.path.dirname( filepath ),
-                             path_dir )
+
+def _GetPathsStandardCase( path_dir, use_working_dir, filepath, working_dir ):
+
+  def GetAbsolutePath( path_dir, use_working_dir, filepath, working_dir ):
```

Comment:
If it doesn't need to be a inner function, make it a top-level one. The reason is that inner functions have access to all the vars from the enclosing function, and are thus worse at encapsulation. Even if the inner function only uses it's own params, the reader has to audit the whole function to figure that out.